crates/core/src/diff.rs

```diff
@@ -20,11 +19,20 @@ fn powersync_diff_impl(
 ) -> Result<String, SQLiteError> {
     let data_old = args[0].text();
     let data_new = args[1].text();
+    let ignore_removed = args.get(2).map_or(false, |v| v.int() != 0);
 
-    diff_objects(data_old, data_new)
+    diff_objects_with_options(data_old, data_new, ignore_removed)
 }
 
-pub fn diff_objects(data_old: &str, data_new: &str) -> Result<String, SQLiteError> {
+/// Returns a JSON object containing entries from [data_new] that are not present in [data_old].
+///
+/// When [ignore_removed_columns] is set, columns that are present in [data_old] but not in
+/// [data_new] will not be present in the returned object. Otherwise, they will be set to `null`.
```

Comment:
It may be useful to describe the specific use case here (calculating the "old" values with `include_old_only_when_changed`).